### PR TITLE
Change Global to Minimize and change type names in accordance

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
PTAL @vladimir-ch 

Fixes #482

**Comment by btracey** [comment]
I tried to organize the commits logically. Some of the documentation changes are smeared between commits as there wasn't a clear break comment wise.

File: optimize/global.go [hunk 1]
```diff
@@ -93,32 +93,36 @@ type Method interface {
...
 //
 // The third argument contains the settings for the minimization. The
-// DefaultGlobalSettings function can be called for a Settings struct with the
-// default values initialized. If settings == nil, the default settings are used.
-// All of the settings will be followed, but many of them may be counterproductive
-// to use (such as GradientThreshold). Global cannot guarantee strict adherence
-// to the bounds specified when performing concurrent evaluations and updates.
+// DefaultLocalSettings and DefaultGlobalSettings functions can be called for
+// different default settings depending on the optimization method. If
+// settings == nil, the default settings are used. All settings will be honored
```
Comment (kortschak) [inline optimize/global.go]: s/==/is/

File: optimize/global.go [hunk 2]
```diff
@@ -93,32 +93,36 @@ type Method interface {
...
+// for all Methods, even if that setting is counter-productive to the method.
+// However, the information used to check the Settings, and the times at which
+// they are checked, are controlled by the Method. For example, if the Method
+// never evaluates the gradient of the function then GradientThreshold will not
+// be checked.  Minimize cannot guarantee strict adherence to the bounds
```
Comment (kortschak) [inline optimize/global.go]: Remove double space after dot.

File: optimize/global.go [hunk 3]
```diff
@@ -93,32 +93,36 @@ type Method interface {
...
 //
-// Global returns a Result struct and any error that occurred. See the
+// Minimize returns a Result struct and any error that occurred. See the
 // documentation of Result for more information.
 //
 // See the documentation for Method for the details on implementing a method.
 //
-// Be aware that the default behavior of Global is to find the minimum.
-// For certain functions and optimization methods, this process can take many
+// Be aware that the default settings of Minimize are to accurately find the
```
Comment (kortschak) [inline optimize/global.go]: "behavior" reads better here IMO.

File: optimize/global.go (hunk 1 above)
Comment (btracey) [reply optimize/global.go]: Done.

File: optimize/global.go (hunk 2 above)
Comment (btracey) [reply optimize/global.go]: Done.

**Comment by btracey** [comment]
Glad you like the changes @kortschak

File: optimize/global.go [hunk 4]
```diff
@@ -20,107 +20,116 @@ func DefaultSettingsGlobal() *Settings {
...
-	// equal to the return from InitGlobal.
-	RunGlobal(operation chan<- GlobalTask, result <-chan GlobalTask, tasks []GlobalTask)
+	// equal to the return from Init.
+	Run(operation chan<- Task, result <-chan Task, tasks []Task)
 }
 
-// Global uses a global optimizer to search for the global minimum of a
-// function. A maximization problem can be transformed into a
-// minimization problem by multiplying the function by -1.
+// Minimize uses an optimizer to search for the minimum of a function. A
```
Comment (vladimir-ch) [inline optimize/global.go]: ... for _a_ minimum ... ?

File: optimize/global.go [hunk 5]
```diff
@@ -20,107 +20,116 @@ func DefaultSettingsGlobal() *Settings {
...
+// Some Methods do not require an initial location, but initX must still be
+// specified for the dimension of the optimization problem.
+//
 // The third argument contains the settings for the minimization. The
-// DefaultGlobalSettings function can be called for a Settings struct with the
-// default values initialized. If settings == nil, the default settings are used.
-// All of the settings will be followed, but many of them may be counterproductive
-// to use (such as GradientThreshold). Global cannot guarantee strict adherence
-// to the bounds specified when performing concurrent evaluations and updates.
+// DefaultLocalSettings and DefaultGlobalSettings functions can be called for
```
Comment (vladimir-ch) [inline optimize/global.go]: DefaultSettingsLocal and DefaultSettingsGlobal

File: optimize/types.go [hunk 6]
```diff
@@ -166,13 +166,10 @@ func (p Problem) satisfies(method Needser) error {
 //
 // If Recorder is nil, no information will be recorded.
 type Settings struct {
-	// InitX specifies an initial location to communicate to the Method. If InitX
-	// is nil, then a slice of zeros is used as a default value.
-	InitX []float64
-	// InitValues specifies properties known at InitX (function value, gradient, etc.).
-	// If InitX is nil, InitValues must be also. If InitValues is non-nil, then
+	// InitValues specifies properties (function value, gradient, etc.). known
```
Comment (vladimir-ch) [inline optimize/types.go]: Please remove dot after `)`.

**Comment by btracey** [comment]
Done. PTAL @vladimir-ch

File: optimize/global.go [hunk 7]
```diff
@@ -20,107 +20,116 @@ func DefaultSettingsGlobal() *Settings {
...
+//
 // The third argument contains the settings for the minimization. The
-// DefaultGlobalSettings function can be called for a Settings struct with the
-// default values initialized. If settings == nil, the default settings are used.
-// All of the settings will be followed, but many of them may be counterproductive
-// to use (such as GradientThreshold). Global cannot guarantee strict adherence
-// to the bounds specified when performing concurrent evaluations and updates.
+// DefaultSettingsLocal and DefaultSettingsGlobal functions can be called for
+// different default settings depending on the optimization method. If
+// settings is nil, the default settings are used. All settings will be honored
```
Comment (sbinet) [inline optimize/global.go]: perhaps mention that the default settings being then used are the DefaultSettingsLocal ?

Comment (btracey) [reply optimize/global.go]: Done.

**Comment by btracey** [comment]
Thanks!

**Pull request merged**